cmd/scheduler/main.go

```diff
@@ -48,7 +48,7 @@ func main() {
 		app.WithPlugin(coscheduling.Name, coscheduling.New),
 		app.WithPlugin(loadvariationriskbalancing.Name, loadvariationriskbalancing.New),
 		app.WithPlugin(noderesources.AllocatableName, noderesources.NewAllocatable),
-		app.WithPlugin(noderesourcetopology.Name, noderesourcetopology.New),
+		app.WithPlugin(filter.Name, filter.New),
```

Comment:
I understand you want to tease apart the filter and score logic, but it doesn't look quite necessary as they share some commons structs and should semantically be placed into the same package IMO.

A reference is the upstream [podtoplogyspread](https://github.com/kubernetes/kubernetes/tree/master/pkg/scheduler/framework/plugins/podtopologyspread) filter and score plugins. The folder can be kept as noderesourcetopology, and you can put the logic into different files: plugin.go, filter.go, score.go, common.go, etc.

Reply:
Thank you @Huang-Wei for taking the time to review this, I see your point but, I tried to implement that in a way that keeps the modularity of the package as much as possible and I'll explain:
The score plugin depends on the filter plugin.
There is no reasonable/realistic scenario to use the score plugin without the filter plugin, so combining the two into one plugin is the right way (as you were suggesting).

On the other hand, the filter plugin not depends on the score plugin. 
A user might use the filter plugin without the score plugin, because, for example, the user prefers not to use the scoring strategies that we offer, but use the default scheduler score plugin instead.

So by combining the two plugins together we enforce using our score strategies, and IMHO it might be less flexible unless you have an idea, or you know a way to avoid that? 

Please let me know what you think

Reply:
Good question. Basically, upon the plugin initialization, it still serves as a combination of Filter and Score plugins. However, users can disable the Score functionalities via disabling it in the ComponentConfig. Here is an example how to disable the upstream NodeResourceFit score plugin:

```yaml
...
profiles:
- plugins:
    score:
      disabled:
      - name: NodeResourcesFit
```

Reply:
BTW: we need to disable one particular score plugin is because there is a default list of enabled plugins. For scheduler-plugins, the enabled plugins need to be always explicitly enabled, so we just specify which Filter/Score plugins to enable in the component config yaml.

Reply:
I changed the package structure per your request.
Please continue with the review when possible and let me know if additional changes are needed.